Upgrade to latest Design System

virginiacc opened the pull request:
Addresses https://github.com/cfpb/design-system-react/issues/417.

Updates DSR to use the latest version of the CFPB Design System and migrates from Less to Sass

## Changes
- Update to latest version of the DS
- Add aliases for cfpb-design-system scss imports in vite config
- Switch from Less to Sass
- Update classes to use revised BEM naming approach
- Remove classes that have been deprecated in the DS, including
   - [narrow content option](https://github.com/cfpb/design-system/pull/1928) (`content_main__narrow`)
   - [bleedbar layout](https://github.com/cfpb/design-system/pull/1954) (`content__bleedbar`)
   - [directory table](https://github.com/cfpb/design-system/pull/2069) (`o-table__entry-header-on-small`)
   - [input with button](https://github.com/cfpb/design-system/pull/1979) (`o-form__input-w-btn`) — replaced with `o-search-input`
   - [list link class](https://github.com/cfpb/design-system/pull/1989) (`m-list__link`)
   - [icon link classes](https://github.com/cfpb/design-system/pull/1987) (`a-link--icon-before-text` and `a-link--icon-after-text`)
- Update tests and stories
- Add postcss icon processor
- Add icon color styles for notifications
- Update vite version

## How to test this PR

1. Run `yarn` and `yarn start` and verify that stories appear as expected
2. Run `yarn test` and verify that tests pass
3. Run visual regression on DSR and SBL and check for breaking changes

BrewCityBoy commented:
Pulled down this update and everything looks to check out. No breaks. No failures.

virginiacc commented:
Thanks for reviewing & identifying those issues, @billhimmelsbach! I think I've fixed all of them for now (although I wouldn't mind revisiting the link components more thoroughly at some point). I also updated to the latest release of the DS -- I don't think there have been any significant changes and didn't see any issues in a quick check of the stories, but I can revert to this PR's original version if that seems less risky.

virginiacc commented:
@billhimmelsbach Oof, yes, I broke the inline link underlining with my last commit -- thanks for catching! That should be fixed now, along with a few other issues I found when reviewing the stories with links.

BrewCityBoy reviewed (commented):
Checked out the branch before pulling the update and applying the changes. Looks like everything checks out @billhimmelsbach.
